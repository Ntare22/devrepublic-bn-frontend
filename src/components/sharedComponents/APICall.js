/* eslint-disable no-alert */
/* eslint-disable no-console */
import React, { useState } from 'react';
import axios from 'axios';

const ApiCall = () => {
  const [message, setMessage] = useState('');
  const handleApiCall = async () => {
    const response = await axios.get('https://devrepublic-bn-backend.herokuapp.com/');
    setMessage(response.data);
  };
  return (
    <div>
      <button type="button" onClick={handleApiCall}>Click</button>
      <h3>{message.message}</h3>
    </div>
  );
};

export default ApiCall;
