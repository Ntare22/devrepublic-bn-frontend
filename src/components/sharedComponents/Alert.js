import React from 'react';
import { Alert } from 'reactstrap';
import PropTypes from 'prop-types';
import '../../scss/login.scss';

const ErrorAlert = ({
  message, visible, textColor, otherClassName,
}) => (
  <Alert color="info" className={`text-${textColor} ${otherClassName}`} isOpen={visible}>
    {message}
  </Alert>
);
ErrorAlert.propTypes = {
  message: PropTypes.string.isRequired,
  visible: PropTypes.bool.isRequired,
  textColor: PropTypes.string.isRequired,
  otherClassName: PropTypes.string,
};
ErrorAlert.defaultProps = {
  otherClassName: '',
};

export default ErrorAlert;
