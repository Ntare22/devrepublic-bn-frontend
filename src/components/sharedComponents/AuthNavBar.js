import React, { Component } from 'react';
import {
  Navbar, Nav, Dropdown, DropdownButton, Button,
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import '../../scss/navbar.scss';
import '../../scss/authNavbar.scss';
import selectLanguage from '../../redux/actions/i18n';
import translate from '../languages/Translate';
import bell from '../../assets/notificationbell.png';
import placeholder from '../../assets/users.png';
import logo from '../../assets/logo.png';

export class AuthNavBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      language: props.language.language,
    };
    this.handleSelect = this.handleSelect.bind(this);
  }

  handleSelect = (evt) => {
    const newString = evt.split('#')[1];
    const actionOutput = this.props.selectLanguage(newString);
    this.setState({ language: actionOutput.payload });
  }

  render() {
    const { language } = this.state;
    const { image } = this.props;
    return (
      <Navbar fixed="top" expand="lg" className="nav-bar">
        <Navbar.Brand href="#home">
          <img
            src={logo}
            width="30"
            height="30"
            className="d-inline-block align-top"
            alt="barefoot nomad logo"
          />
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" className="nav-bar_items" bsPrefix="navbar-toggler" />
        <Navbar.Collapse id="responsive-navbar-nav" bsPrefix="navbar-collapse">
          <Nav className="mr-auto" />
          <Nav>
            <Button className="nav-bar_btn">
              {' '}
              {
                  translate('HOME')
                }
            </Button>
            <Nav.Link>
              <img src={bell} alt="bell" className="notification-bell" />
            </Nav.Link>
            <img src={image || placeholder} alt="profile" className="nav-profile-img" />
            <DropdownButton title={language} alignRight variant="inherit" onSelect={this.handleSelect} className="select-lang" bsPrefix="nav-bar_dropdown">
              <Dropdown.Item className="lang-item" href="#en">ENGLISH</Dropdown.Item>
              <Dropdown.Item className="lang-item" href="#fr">FRANCAIS</Dropdown.Item>
            </DropdownButton>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

AuthNavBar.propTypes = {
  language: PropTypes.object,
  selectLanguage: PropTypes.func.isRequired,
  image: PropTypes.string,
};
const MapStateToProps = ({ language, profile }) => ({
  language,
  image: profile.image,
});
export default connect(MapStateToProps, { selectLanguage })(AuthNavBar);
