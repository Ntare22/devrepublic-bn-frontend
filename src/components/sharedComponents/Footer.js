import React from 'react';
import {
  Row, Col, Container,
} from 'react-bootstrap';
import '../../scss/footer.scss';
import translate from '../languages/Translate';

const Footer = () => (
  <Container className="fixed-bottom footer" fluid>
    <Row className="justify-content-md-center">
      <Col md="auto" className="footer_text">
        { translate('Copyright 2020 devrepublic team. All rights reserved.')}
      </Col>
    </Row>
  </Container>
);

export default Footer;
