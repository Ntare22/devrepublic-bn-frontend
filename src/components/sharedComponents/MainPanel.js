import React, { Component } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import '../../scss/mainpanel.scss';
import PropTypes from 'prop-types';
import pagination from '../../util/pagination';

export default class MainPanel extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filter: 'all',
      currentPage: 1,
    };
  }

  componentDidUpdate(prevProp, prevState) {
    if (prevState.filter !== this.state.filter || this.state.currentPage !== prevState.currentPage) {
      let newArray;
      newArray = this.props.allTrips.filter((el) => el.status === this.state.filter);
      if (this.state.filter === 'all') newArray = this.props.allTrips;
      this.props.filter(pagination(newArray, this.state.currentPage));
    }
  }

  handleSelect = (eventKey) => {
    const { value } = eventKey.target;
    this.setState({ filter: value, currentPage: 1 });
  };

  pageUp = () => {
    this.setState((prevState, props) => {
      const newPage = prevState.currentPage + 1;
      let totalPage;
      totalPage = props.allTrips.filter((el) => el.status === prevState.filter).length / 5;
      if (prevState.filter === 'all') totalPage = props.allTrips.length / 5;
      if (newPage > Math.ceil(totalPage)) {
        return { currentPage: Math.ceil(totalPage) || 1 };
      }
      return { currentPage: newPage };
    });
  }

  pageDown = () => {
    if (this.state.currentPage === 1) {
      return;
    }
    this.setState((prevState) => ({ currentPage: prevState.currentPage - 1 }));
  }

  render() {
    const { currentPage, filter } = this.state;
    return (
      <Container fluid>
        <Row>
          <Col xs="12" sm="12" md="12" lg="8">
            <div className="pagination">
              <div className="pagination_item">
                <div className="select-container">
                  <select className="pagination_select" name="filter" onChange={this.handleSelect} value={filter}>
                    <option value="all">All Travel Requests</option>
                    <option value="open">Open travel Requests</option>
                    <option value="rejected">Rejected Travel Requests</option>
                    <option value="accepted">Accepted Travel Requests</option>
                  </select>
                  <i className="arrow down" />
                </div>
              </div>
              <div className="page-controller">
                <div className="pagination_item number">
                  <h5>
                    {currentPage}
                  </h5>
                </div>
                <div className="pagination_item arrows">
                  <button type="button" name="down" onClick={this.pageDown}><i className="arrow left"> </i></button>
                  <button type="button" onClick={this.pageUp}><i className="arrow right"> </i></button>
                </div>
              </div>
            </div>
          </Col>
        </Row>
      </Container>
    );
  }
}
MainPanel.propTypes = {
  allTrips: PropTypes.array,
  filter: PropTypes.func,
};
