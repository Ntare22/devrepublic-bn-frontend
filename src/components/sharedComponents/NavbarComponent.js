import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Navbar, Nav, Dropdown, DropdownButton, Button,
} from 'react-bootstrap';
import '../../scss/navbar.scss';
import logo from '../../assets/logo.png';
import { selectLanguage } from '../../redux/actions/i18n';
import translate from '../languages/Translate';

class NavBarComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      language: this.props.language.language,
    };
  }

  handleSelect = (evt) => {
    const newString = evt.split('#')[1];
    const actionOutput = this.props.selectLanguage(newString);
    this.setState({ language: actionOutput.payload });
  }

  render() {
    const { language } = this.state;
    return (
      <Navbar fixed="top" expand="lg" className="nav-bar">
        <Navbar.Brand href="#home">
          <img
            src={logo}
            width="30"
            height="30"
            className="d-inline-block align-top"
            alt="barefoot nomad logo"
          />
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" className="nav-bar_items" bsPrefix="navbar-toggler" />
        <Navbar.Collapse id="responsive-navbar-nav" bsPrefix="navbar-collapse">
          <Nav className="mr-auto" />
          <Nav>
            <Nav.Link href="/login" className="nav-bar_item">
              {
                  translate('Log In')
                }
            </Nav.Link>
            <Button className="nav-bar_btn">
              {
                   translate('Register')
                 }
            </Button>

            <DropdownButton className="test" title={language} onSelect={this.handleSelect} alignRight variant="inherit" bsPrefix="nav-bar_dropdown">
              <Dropdown.Item className="lang-item" href="#en">ENGLISH</Dropdown.Item>
              <Dropdown.Item className="lang-item" href="#fr">FRANCAIS</Dropdown.Item>
            </DropdownButton>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}
const MapStateToProps = ({ language }) => ({
  language,
});
export default connect(MapStateToProps, { selectLanguage })(NavBarComponent);
export { NavBarComponent };

NavBarComponent.propTypes = {
  language: PropTypes.object,
  selectLanguage: PropTypes.func,

};
