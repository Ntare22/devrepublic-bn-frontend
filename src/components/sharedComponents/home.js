import React from 'react';
import '../../scss/app.scss';

const Home = () => <h1 className="welcome"> welcome to barefoot </h1>;

export default Home;
