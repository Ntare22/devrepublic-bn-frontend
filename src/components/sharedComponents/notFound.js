import React from 'react';

const NotFound = () => <div className="container"><h1> Oop! 404 Page not found </h1></div>;

export default NotFound;
