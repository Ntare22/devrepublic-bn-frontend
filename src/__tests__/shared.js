import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from '../components/sharedComponents/Footer';
import NotFound from '../components/sharedComponents/notFound';
import Home from '../components/sharedComponents/home';
import { NavBarComponent as NavBarCompo } from '../components/sharedComponents/NavbarComponent';

describe('Not found page Tests', () => {
  const wrapper = shallow(<NotFound />);
  it('Should Render footer component', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
describe('Home page Tests', () => {
  const wrapper = shallow(<Home />);
  it('Should Render footer component', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
describe('Footer Tests', () => {
  const wrapper = shallow(<Footer />);
  it('Should Render footer component', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('Should conatain footer class', () => {
    expect(wrapper.find('.fixed-bottom').length).toBe(1);
  });
});
describe('Navbar class component snapshot', () => {
  const render = (fn = shallow) => {
    const defaultProps = {
      langauge: '',
      selectLanguage: jest.fn().mockReturnValue({ payload: '' }),
    };
    return fn(
      <NavBarCompo {...defaultProps} />,
    );
  };
  it('should create a snapshot', () => {
    expect(toJson(render)).toMatchSnapshot();
  });
});
