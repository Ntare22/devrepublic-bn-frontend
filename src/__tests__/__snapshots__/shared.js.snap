// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar class component snapshot should create a snapshot 1`] = `undefined`;
